src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs

```diff
@@ -72,24 +72,36 @@ public class WebResponseObject
 
         #endregion Properties
 
+        #region Protected Fields
+
+        /// <summary>
+        /// Time permitted between reads or Timeout.InfiniteTimeSpan for no timeout.
+        /// </summary>
+        protected TimeSpan perReadMillisecondsTimeout;
```

Comment:
@stevenebutler why did you choose `perReadMillisecondsTimeout` instead of something like `_perReadTimeout` ?
Maybe you can cache `perReadTimeout = NetworkTimeout` directly in `WebRequestPSCmdlet.Common.cs`

Reply:
Even if cached in the cmdlet, it would still need to be passed down into this class because it needs to know the value.

I used a getter to convert the value in the cmdlet and came locally when used multiple times but could change it to a pre calculated value if you think it's worthwhile. I did consider this but didn't think it would be of much benefit.

Reply:
I have renamed the protected property to perReadTimeout but I won't cache the TimeSpan value in the web request unless someone feels strongly this should be done.